server/src/routes/user.ts

```diff
@@ -0,0 +1,39 @@
+import express, { Request, Response, Router } from "express"
+
+const router: Router = express.Router()
+
+router.get('/:uid', async (req: Request, res: Response) => {
```

Comment:
This can be simpler, we don't need to check `if(uid)` since URL parameters are always present if the route matches:

```
router.get('/:uid', async (req: Request, res: Response) => {
  const { uid } = req.params;
  
  // URL parameters are always strings, so if it exists in the URL, uid will never be falsy
  // We only need to check if we want specific uid formats (like UUID)
  
  try {
    res.status(200).json({ 
      message: `Fetching user with ID: ${uid}` 
    });
  } catch (error) {
    res.status(400).json({ 
      message: `Unable to fetch user with ID: ${uid}`,
      error: error.message 
    });
  }
});
```